gpkit/solution_array.py

```diff
@@ -231,14 +231,18 @@ def results_table(data, title, minval=0, printunits=True, fixedcols=True,
             lines.append([varstr, valstr, units, label])
         else:
             varstr = "$%s$" % varstr.replace(" : ", "")
+            boxunit = var.unitstr()
+            if not boxunit:
+                boxunit = "-"
             if latex == 1:  # normal results table
-                lines.append([varstr, valstr, "$%s$" % var.unitstr(), label])
+                lines.append([varstr, valstr, "$\\left[%s\\right]$" % boxunit,
+                              label])
                 coltitles = [title, "Value", "Units", "Description"]
             elif latex == 2:  # no values
-                lines.append([varstr, "$%s$" % var.unitstr(), label])
+                lines.append([varstr, "$\\left[%s\\right]$" % boxunit, label])
                 coltitles = [title, "Units", "Description"]
             elif latex == 3:  # no description
-                lines.append([varstr, valstr, "$%s$" % var.unitstr()])
+                lines.append([varstr, valstr, "$\\left[%s\\right]$" % boxunit])
```

Comment:
I'm not a fan of putting brackets around units to be printed in a table -- again would prefer to let rogue users do that themselves  :P